ush/python/pygfs/task/aero_emissions.py

```diff
@@ -37,48 +44,439 @@ def __init__(self, config: Dict[str, Any]) -> None:
+
+    @staticmethod
+    @logit(logger)
+    def open_qfed(fname: Union[str, os.PathLike], out_vars: list = None, input_vars: list = None) -> xr.Dataset:
+        """
+        Open QFED2 fire emissions data and renames variables to a standard (using the GBBEPx names to start with).
 
         Parameters
         ----------
-        workdir : str | os.PathLike
-            work directory with the staged data, parm files, namelists, etc.
-        aprun_cmd : str
-            launcher command for executable.x
+        fname : str or list of str
+            Path(s) to the QFED2 fire emissions files
 
         Returns
         -------
+        xr.Dataset
+            Dataset containing the fire emissions data
+        """
+        vrs = out_vars  # ["BC", "CH4", "CO", "CO2", "NH3", "NOx", "OC", "PM2.5", "SO2"]
+        qfed_vars = input_vars  # ["bc", "ch4", "co", "co2", "nh3", "no", "oc", "pm25", "so2"]
+
+        if len(fname) > 1:
+            files = np.sort(fname)
+        else:
+            files = np.sort(glob(fname))
+
+        try:
+            if files.size == 0:
+                raise Exception("FATAL ERROR: No QFED files were found")
+        except Exception as ee:
+            logger.exception("FATAL ERROR: QFED files not found.")
+            raise Exception(f"FATAL ERROR: Unable to find files, ABORT")
+        found_species = []
+        dset_dict = {}
+        for f in files:
+            index_good = [[i, v] for i, v in enumerate(qfed_vars) if v in f]
+            good = index_good[0][0]
+            found_species.append(index_good[0][1])
+
+            try:
+                logger.info("Opening QFED file: {filename}".format(filename=f))
+                with xr.open_dataset(f, decode_cf=False).biomass as da:
+                    da.name = vrs[good]
+                    dset_dict[vrs[good]] = da
+            except Exception as ee:
+                logger.exception("FATAL ERROR: unable to read dataset {error}".format(error=ee))
+                raise Exception("FATAL ERROR: Unable to read dataset, ABORT!")
+
+        dset = xr.Dataset(dset_dict)
+        return dset
+
+    @staticmethod
+    @logit(logger)
+    def open_climatology(fname: list = None) -> xr.Dataset:
+        """
+        Open climatology files and concatenate them along the time dimension.
+
+        Parameters:
+        -----------
+        fname: str or list of str
+            Path(s) to the climatology files.
+
+        Returns:
+        --------
+            xr.Dataset: Concatenated dataset containing the climatology data.
+        """
+        # array to house datasets
+        das = []
+
+        if len(fname) > 1:
+            files = np.sort(fname)
+        else:
+            files = np.sort(glob(fname))
+        
+        logger.info("Process Climatlogy Files")
+        logger.info("  Opening Climatology File: {filename}".format(filename=fname[0]))
+        xr.open_dataset(files[0])
+        for i, f in enumerate(files):
```

Comment:
The index isn't being used, I don't think you need enumerate
```suggestion
        for filename in files:
```